cloud/scope/powervs_cluster.go

```diff
@@ -1125,6 +1127,7 @@ func (s *PowerVSClusterScope) ReconcileVPCSubnets() (bool, error) {
 				return false, fmt.Errorf("failed to get VPC subnet with ID %s", *subnetID)
 			}
 			// check for next subnet
+			s.SetVPCSubnetStatus(*subnetDetails.Name, infrav1beta2.ResourceReference{ID: subnetDetails.ID})
```

Comment:
This was added mainly to handle the scenario where existing resource ID is passed in spec. Might not be the best way but found this easier as of now with the current code flow as changes otherwise was making the code complicated

Reply:
When id is set in spec, why do you think we need to set it in status as well? Do we follow same thing for other resources as well.

Reply:
Currently, we are following this for all resources. As `controllerCreated` will be false in this case, ideally, we wouldn't need ID which would be used during deletion.
Should we consider setting ID only when name is passed?